plugin/src/main/java/io/fabric8/maven/plugin/mojo/develop/DebugMojo.java

```diff
@@ -251,6 +273,23 @@ private boolean enableDebugging(HasMetadata entity, PodTemplateSpec template) {
                         container.setPorts(ports);
                         enabled = true;
                     }
+                    if (debugSuspend) {
+                        // Setting a random value to force pod restart
+                        this.debugSuspendValue = String.valueOf(new Random().nextLong());
+                        KubernetesResourceUtil.removeEnvVar(env, DebugConstants.ENV_VAR_JAVA_DEBUG_SUSPEND);
+                        KubernetesResourceUtil.setEnvVar(env, DebugConstants.ENV_VAR_JAVA_DEBUG_SUSPEND, this.debugSuspendValue);
+                        container.setEnv(env);
+                        if (container.getReadinessProbe() != null) {
+                            log.info("Readiness probe will be disabled on " + getKind(entity) + " " + getName(entity) + " to allow attaching a remote debugger during suspension");
+                            container.setReadinessProbe(null);
+                        }
+                        enabled = true;
+                    } else {
+                        if (KubernetesResourceUtil.removeEnvVar(env, DebugConstants.ENV_VAR_JAVA_DEBUG_SUSPEND)) {
```

Comment:
Could you always set the variable with 'true', 'false' (and evaluate it accordingly in the startup script). This would simplify the code here (you don't need an "if" then, but set it always) and it makes it more explicit for the user within the container whether suspending is switched on or off (since the env var is always set).